src/DataCollection.Shared/Binding Support/IdentifyController.cs

```diff
@@ -108,79 +104,97 @@ private void MapView_DoubleTapped(object sender, GeoViewInputEventArgs e)
+                    var identifyResult = await mapView.IdentifyGraphicsOverlayAsync(targetOverlay, tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
+                    if (_currentSource.IsCancellationRequested)
                     {
-                        // identify features in the target layer, passing the tap point, tolerance, types to return, and max results
-                        var identifyResult = await mapView.IdentifyGraphicsOverlayAsync(targetOverlay, tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
-                        graphicsOverlayResults = new List<IdentifyGraphicsOverlayResult> { identifyResult }.AsReadOnly();
+                        return;
                     }
-                    else if (target is ArcGISSublayer sublayer)
-                    {
-                        var layer = mapView?.Map?.AllLayers?.OfType<ArcGISMapImageLayer>()?.Where(l => l.Sublayers.Contains(sublayer))?.FirstOrDefault();
+                    graphicsOverlayResults = new List<IdentifyGraphicsOverlayResult> { identifyResult }.AsReadOnly();
+                }
+                else if (target is ArcGISSublayer sublayer)
+                {
+                    var layer = mapView?.Map?.AllLayers?.OfType<ArcGISMapImageLayer>()?.Where(l => l.Sublayers.Contains(sublayer))?.FirstOrDefault();
 
-                        if (layer != null)
-                        {
+                    if (layer != null)
+                    {
 
-                            // identify features in the target layer, passing the tap point, tolerance, types to return, and max results
-                            var topLevelIdentifyResult = await mapView.IdentifyLayerAsync(layer, tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount);
-                            var sublayerIdentifyResult = topLevelIdentifyResult?.SublayerResults?.Where(r => r.LayerContent.Equals(sublayer)).FirstOrDefault();
+                        // identify features in the target layer, passing the tap point, tolerance, types to return, and max results
+                        var topLevelIdentifyResult = await mapView.IdentifyLayerAsync(layer, tapScreenPoint, pixelTolerance, returnPopupsOnly, maxResultCount, _currentSource.Token);
+                        var sublayerIdentifyResult = topLevelIdentifyResult?.SublayerResults?.Where(r => r.LayerContent.Equals(sublayer)).FirstOrDefault();
 
-                            layerResults = new List<IdentifyLayerResult> { sublayerIdentifyResult }.AsReadOnly();
-                        }
+                        layerResults = new List<IdentifyLayerResult> { sublayerIdentifyResult }.AsReadOnly();
                     }
-
-                    OnIdentifyCompleted(layerResults, graphicsOverlayResults);
                 }
+                OnIdentifyCompleted(layerResults, graphicsOverlayResults);
```

Comment:
Should we check `_currentSource.IsCancellationRequested` before we get here? Otherwise we could get overlay identify results even though the operation was cancelled.